src/perform_step/rkc_perform_step.jl

```diff
@@ -296,3 +296,169 @@ end
   integrator.k[2] = integrator.fsallast
   integrator.u = u
 end
+
+
+
+function initialize!(integrator, cache::RKCConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+@muladd function perform_step!(integrator, cache::RKCConstantCache, repeat_step=false)
+  @unpack t, dt, uprev, u, f, p, fsalfirst = integrator
+  maxeig!(integrator, cache)
+  # The the number of degree for Chebyshev polynomial
+  maxm = max(2,Int(floor(sqrt(integrator.opts.internalnorm(integrator.opts.reltol)/(10.0*eps(integrator.opts.internalnorm(uprev)))))))
+  mdeg = 1 + Int(floor(sqrt(1.54*dt*integrator.eigen_est + 1.0)))
+  if mdeg >= maxm
+    mdeg = maxm
+  end
+
+  w0 = 1.0 + 2.0/(13.0*(mdeg^2.0))
+  temp1 = w0^2.0 - 1.0
+  temp2 = sqrt(temp1)
+  arg   = mdeg*log(w0 + temp2)
+  w1    = (sinh(arg)*temp1) / (cosh(arg)*mdeg*temp2 - w0*sinh(arg))
+  b1    = 1.0/((2.0*w0)^2.0)
+  b2    = b1
+
+  # stage-1
+  gprev2 = copy(uprev)
+  μs     = w1*b1
+  gprev  = uprev + dt*μs*fsalfirst
+  th2  = 0.0
```

Comment:
this code is not type safe. These all assume Float64. Use `zero(eltype(u))` and `one(eltype(u))` instead